src/interface/state_descriptor.cpp

```diff
@@ -566,4 +566,37 @@ StateDescriptor::GetVariableNames(const Metadata::FlagCollection &flags) {
   return GetVariableNames({}, flags, {});
 }
 
+// Get the total length of this StateDescriptor's variables when packed
+int StateDescriptor::GetPackDimension(const std::vector<std::string> &req_names,
+                                      const Metadata::FlagCollection &flags,
+                                      const std::vector<int> &sparse_ids) {
+  std::vector<std::string> names = GetVariableNames(req_names, flags, sparse_ids);
+  int nvar = 0;
+  for (auto name : names) {
+    const auto &meta = metadataMap_[VarID(name)];
+    int var_len = 0;
+    if (meta.Shape().size() < 1) {
+      var_len = 1;
+    } else {
+      var_len = meta.Shape()[0];
```

Comment:
Does this work for tensor variables?

Reply:
Indeed it did not!  Should be fixed w/regression test now.

Reply:
Can you not do this with an (almost) one-liner rather than all this branching?
```C++
if (meta.Shape().size() > 3) {
  PARTHENON_THROW("Computing total lengths of tensors of rank >3 is not supported!");
} else {
  // if meta.Shape().size() < 1, then var_len will be set to the initialization value, which is 1.
  var_len = std::accumulate(meta.Shape().begin(), meta.Shape.end(), 1, [](auto a, auto b){ return a*b;});
}
```

Reply:
Sure! Since the `accumulate` version is fully general, I just removed the error and made it just one line.  I don't think the branching matters since this function is called at most a few times, but having a general version we don't have to think about again will be nice.

Reply:
:+1: I don't think the branching matters for performance but the accumulate one-liner is much easier to read IMO.